### PR TITLE
feat: enable istio injection

### LINKED ISSUE #6067: Istio injection always disabled
While trying to put together a Kubeflow distribution that includes as many as possible elements inside the Istio mesh, we found that for Kubeflow Pipelines steps the istio sidecar injection is always overriden to "false". 

(This would for example make it impossible for a pipelines step to make calls to a KFServing model in the same namespace which has an istio sidecar enabled).

This seems to be hardcoded here:

https://github.com/kubeflow/pipelines/blob/fff62e1521e89dd31c87eec03073b8f8627d73a3/backend/src/apiserver/resource/resource_manager.go#L384

I presume that was done because the Pipelines owner are very sure that KFP would not work with injection enabled. As far as I know, newer versions of Argo Workflows should now support it.

Could we turn this into an ENV var?

### PR DESCRIPTION
**Pull request by capri-xiyue**
**Description of your changes:**
Fixed #6067 
Enable istio injection

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: backend/src/apiserver/resource/test_util.go [hunk 1]
```diff
@@ -4,7 +4,7 @@ import "github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1"
 
 func AddRuntimeMetadata(wf *v1alpha1.Workflow) {
 	template := wf.Spec.Templates[0]
-	template.Metadata.Annotations = map[string]string{"sidecar.istio.io/inject": "false"}
+	template.Metadata.Annotations = map[string]string{"sidecar.istio.io/inject": "true"}
```
Comment (Bobgy) [inline backend/src/apiserver/resource/test_util.go:7]: adding the false annotation explicitly disables injection.
but we don't need to make this true though, simply removing the annotation makes most istio envs inject the sidecar by default.

**Review by Bobgy — COMMENTED** [review]
built api server image: gcr.io/ml-pipeline-test/46ff4cbd4cef09725861f866327ec49729ef2775/api-server

@karlschriek you can try this image on your cluster with istio. Can KFP pipelines run properly?

**Comment by Bobgy** [comment]
/hold
wait for more feedback, whether istio sidecar works with argo or not

**Comment by MatthiasCarnein** [comment]
@Bobgy I can confirm that istio-sidecars are injected into pipeline pods with the above image.

With pns and docker executor, everything seems to work and pipelines finish successfully. With emissary executor, pipeline steps never terminate because the wait container cannot be killed. This is probably in line with what @alexec documented. 

Unfortunately, none of these executors work when using istio CNI. Pipeline steps can write data to minio. However, subsequent pipeline steps fail since the init container cannot load the data from minio. Presumably, the init container is not part of the istio mesh and therefore cannot connect to minio. I think this is a common issue with istio and init containers.

**Comment by alexec** [comment]
The recommendation from Argo is to disable ISTIO by default, because of the problems it causes users.

**Comment by alexec** [comment]
An aside - if we can determine the correct kill command for ISTIO - then the pod can be annotated with that command - and Argo will use it to kill ISTIO.

**Comment by Bobgy** [comment]
https://github.com/istio/istio/issues/11659#issuecomment-544432185

**Comment by Bobgy** [comment]
I believe this should be superceded by https://github.com/kubeflow/pipelines/pull/6391, which allows users to optionally enable istio sidecars on some steps, while keeping the default to be disabling istio sidecars.

Given the fact that sometimes istio sidecars work and sometimes not

**Pull request closed without merging**